offline/cable_cru_TRENDY.F90

```diff
@@ -1,1318 +1,1069 @@
+  FillLatitudes: DO LonIndx = 1, xDimSize
+    Lat_all(LonIndx, :) = Latitudes
+  END DO FillLatitudes
+
+  FillLongitudes: DO LatIndx = 1, yDimSize
+    Lon_all(:, LatIndx) = Longitudes
+  END DO FillLongitudes
+
+  ! Some CABLE time units?
+  shod = 0.
+  sdoy = 1
+  smoy = 1
+  syear = CRU%CYEAR
+
+  ! Allocate memory for the average long-wave radiation down for some reason?
+  ALLOCATE(CRU%AVG_LWDN(mLand))
+END SUBROUTINE read_landmask
+
+!!------------------------------------------------------------------------------!
+
+SUBROUTINE prepare_temporal_dataset(FileName, TargetArray)
+  !*## Purpose
+  !
+  ! Read a temporally varying dataset that exists as a text file.
+  !
+  !## Method
+  !
+  ! Read the ```FileName``` text file which contains the yearly CO2 data in
+  ! ```YEAR VALUE``` pairs into ```TargetArray```.
+
+  ! If we have a dataset that's only varying in time, then we should be able to
+  ! just the whole thing into memory immediately.
+
+  CHARACTER(LEN=256), INTENT(IN)    :: FileName
+  REAL, DIMENSION(:), ALLOCATABLE, INTENT(OUT)  :: TargetArray
+
+  ! A unique file ID
+  INTEGER   :: FileID
+
+  ! A placeholder for the line, that we inspect to check for comment lines
+  CHARACTER(LEN=256)    :: LineInFile
+
+  ! Placeholders for the Key in the line, and years which form the start and end
+  ! indexes of the array
+  INTEGER               :: Time, StartTime, EndTime
+  REAL                  :: DummyValue
+
+  ! The status holder, and the file and header counters
+  INTEGER :: ios, LineCounter, HeaderCounter, iter
+
+  ! We're going to assume that we only have a single file.
+  ! We need to iterate through the file twice, first to inspect the number of
+  ! entries in the file to allocate the correct amount of memory, second to
+  ! actually write the data to the array.
+  CALL get_unit(FileID)
+  OPEN(FileID, FILE = FileName, STATUS = "old", ACTION = "read", IOSTAT = ios)
+  IF (ios < 0) THEN
+    WRITE(*,*) "Open of temporal dataset file failed with status:", ios
+    CALL EXIT(5)
+  END IF
+
+  LineCounter = 0
+  HeaderCounter = 0
+
+  DetermineSize: DO
+    ! Read line by line, checking for a header line
+    READ(FileID, '(A)', IOSTAT = ios) LineInFile
+
+    IF (ios < 0) THEN
+      ! Read completed successfully and hit EOF
+
+      ! Take the current time value and set it as the end index of the array
+      EndTime = Time
+
+      EXIT DetermineSize
+    END IF
+
+    ! Check for comment characters- assume they're all at the top of the file
+    IF ((LineInFile(1:1) == "#") .OR. (LineInFile(1:1) == "!")) THEN
+      HeaderCounter = HeaderCounter + 1
+    ELSE
+      ! Otherwise, its a line of useful data
+      LineCounter = LineCounter + 1
+
+      ! Read the line of data as a (time, value) pair. We don't want to use the
+      ! value this time around because we haven't yet allocated the array to
+      ! store it.
+      READ(LineInFile, FMT = '(I4, F)') Time, DummyValue
+    END IF
```

Comment:
The Fortran standard does not allow simply 'F' as format. I propose to use free format because Fortran reads the correct type and precision due to the variable definition:
```
READ(LineInFile, *) Time, DummyValue
```